x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ssl/StoreKeyConfig.java

```diff
@@ -167,30 +167,27 @@ private void checkKeyStore(KeyStore keyStore) throws KeyStoreException {
             "the configured PKCS#11 token does not contain a private key entry";
         throw new IllegalArgumentException(message);
     }
+
     @Override
     public boolean equals(Object o) {
-        if (this == o) return true;
-        if (o == null || getClass() != o.getClass()) return false;
-
-        StoreKeyConfig that = (StoreKeyConfig) o;
-
-        if (keyStorePath != null ? !keyStorePath.equals(that.keyStorePath) : that.keyStorePath != null) return false;
-        if (keyStorePassword != null ? !keyStorePassword.equals(that.keyStorePassword) : that.keyStorePassword != null)
-            return false;
-        if (keyStoreAlgorithm != null ? !keyStoreAlgorithm.equals(that.keyStoreAlgorithm) : that.keyStoreAlgorithm != null)
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
             return false;
-        if (keyPassword != null ? !keyPassword.equals(that.keyPassword) : that.keyPassword != null) return false;
-        return trustStoreAlgorithm != null ? trustStoreAlgorithm.equals(that.trustStoreAlgorithm) : that.trustStoreAlgorithm == null;
+        }
+        StoreKeyConfig that = (StoreKeyConfig) o;
+        return Objects.equals(keyStorePath, that.keyStorePath)
+            && Objects.equals(keyStoreType, that.keyStoreType)
```

Comment:
The field `keyStoreType` was accidentally omitted from `equals()` and `hashCode()` when it was added. I asked the Security team and they said just to add it.